src/ocamlorg/lib/opam_repository.ml

```diff
@@ -86,10 +85,10 @@ let list_package_versions package =
   ls_dir Fpath.(to_string (clone_path / "packages" / package))
 
 let process_opam_file f =
-  let ic = open_in (Fpath.to_string f) in
-  let result = OpamFile.OPAM.read_from_channel ic in
-  close_in ic;
-  result
+  let open Lwt.Syntax in
```

Comment:
Any reason to switch to Lwt? I'm not sure we're gaining anything and this will require all the functions that want to use it to wrap their return type in Lwt.t

Reply:
I think this is required because these changes move the package dependency reading and calculations into a "yieldable" fold over the opam files (the `Lwt.pause` in `Lwt_fold`). We don't want to do blocking I/O here... that is if I understood it correctly :))

Reply:
Indeed the goal is to avoid blocking IO, because any blocking computation would make the server unresponsive for a potentially long time (like the 30 seconds of the `update` call). 
Switching to `Lwt.t` for such computations allow to interleave the 30 seconds workload with other operations from the dream server, such as answering a request.